// ngx-wooapi/bundles/ngx-wooapi.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('rxjs'), require('@angular/common/http'), require('rxjs/operators')) :
    typeof define === 'function' && define.amd ? define('ngx-wooapi', ['exports', '@angular/core', 'rxjs', '@angular/common/http', 'rxjs/operators'], factory) :
    (factory((global['ngx-wooapi'] = {}),global.ng.core,global.rxjs,global.ng.common.http,global.rxjs.operators));
}(this, (function (exports,core,rxjs,http,operators) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var WoocommerceHelperService = (function () {
        function WoocommerceHelperService() {
        }
        /**
         * @param {?} error
         * @return {?}
         */
        WoocommerceHelperService.prototype.handleError = /**
         * @param {?} error
         * @return {?}
         */
            function (error) {
                if (error.error instanceof ErrorEvent) {
                    // A client-side or network error occurred. Handle it accordingly.
                    console.error('An error occurred:', error.error);
                }
                else {
                    // The backend returned an unsuccessful response code.
                    // The response body may contain clues as to what went wrong,
                    console.log('Backend response', error);
                    return rxjs.Observable.throw(error.error);
                }
                // return an observable with a user-facing error message
                return rxjs.Observable.throw({ message: 'Something bad happened; please try again later.' });
            };
        /**
         * @param {?=} query
         * @return {?}
         */
        WoocommerceHelperService.prototype.includeQuery = /**
         * @param {?=} query
         * @return {?}
         */
            function (query) {
                if (query === void 0) {
                    query = {};
                }
                var /** @type {?} */ queryPatch = {};
                Object.keys(query).forEach(function (key) {
                    queryPatch[key] = query[key].toString();
                });
                return queryPatch;
            };
        /**
         * @param {?=} query
         * @return {?}
         */
        WoocommerceHelperService.prototype.includeEncoded = /**
         * @param {?=} query
         * @return {?}
         */
            function (query) {
                if (query === void 0) {
                    query = {};
                }
                var /** @type {?} */ params = new FormData();
                Object.keys(query).forEach(function (key) {
                    params.append(key, query[key]);
                });
                console.log(params);
                return params;
            };
        /**
         * @param {?} response
         * @param {?=} responseBodyKey
         * @return {?}
         */
        WoocommerceHelperService.prototype.includeResponseHeader = /**
         * @param {?} response
         * @param {?=} responseBodyKey
         * @return {?}
         */
            function (response, responseBodyKey) {
                var /** @type {?} */ headers = {};
                response.headers.keys().forEach(function (key) {
                    headers[key] = response.headers.get(key);
                });
                var /** @type {?} */ responseData = {};
                if (responseBodyKey) {
                    responseData[responseBodyKey] = response.body;
                }
                else {
                    responseData = response.body;
                }
                responseData['headers'] = headers;
                return responseData;
            };
        WoocommerceHelperService.decorators = [
            { type: core.Injectable },
        ];
        /** @nocollapse */
        WoocommerceHelperService.ctorParameters = function () { return []; };
        return WoocommerceHelperService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    /** @enum {string} */
    var ProductOrder = {
        "desc": 'desc',
        "asc": 'asc',
    };
    /** @enum {string} */
    var ProductOrderBy = {
        "date": 'date',
        "id": 'id',
        "include": 'include',
        "title": 'title',
        "slug": 'slug',
    };

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var WoocommerceProductsService = (function () {
        function WoocommerceProductsService(httpClient, wooHelper) {
            this.httpClient = httpClient;
            this.wooHelper = wooHelper;
        }
        /**
         * Create a Product
         * @param payload: Product
         */
        /**
         * Create a Product
         * @param {?} payload
         * @return {?}
         */
        WoocommerceProductsService.prototype.createProduct = /**
         * Create a Product
         * @param {?} payload
         * @return {?}
         */
            function (payload) {
                var _this = this;
                return this.httpClient.post("products", payload)
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        /**
         * Retrieve a product
         */
        /**
         * Retrieve a product
         * @param {?} id
         * @return {?}
         */
        WoocommerceProductsService.prototype.retrieveProduct = /**
         * Retrieve a product
         * @param {?} id
         * @return {?}
         */
            function (id) {
                var _this = this;
                return this.httpClient.get("products/" + id)
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        /**
         * Retrieve list of product
         */
        /**
         * Retrieve list of product
         * @param {?=} query
         * @return {?}
         */
        WoocommerceProductsService.prototype.retrieveProducts = /**
         * Retrieve list of product
         * @param {?=} query
         * @return {?}
         */
            function (query) {
                var _this = this;
                return this.httpClient.get("products", { params: this.wooHelper.includeQuery(query), observe: 'response' })
                    .pipe(operators.map(function (value) { return _this.wooHelper.includeResponseHeader(value, 'products'); }), operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        /**
         * Update Product
         */
        /**
         * Update Product
         * @param {?} id
         * @param {?} payload
         * @return {?}
         */
        WoocommerceProductsService.prototype.updateProduct = /**
         * Update Product
         * @param {?} id
         * @param {?} payload
         * @return {?}
         */
            function (id, payload) {
                var _this = this;
                return this.httpClient.put("products/" + id, payload)
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        /**
         * Update Product
         */
        /**
         * Update Product
         * @param {?} id
         * @return {?}
         */
        WoocommerceProductsService.prototype.deleteProduct = /**
         * Update Product
         * @param {?} id
         * @return {?}
         */
            function (id) {
                var _this = this;
                return this.httpClient.delete("products/" + id)
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        /**
         * Retrieve product reviews by product id
         */
        /**
         * Retrieve product reviews by product id
         * @param {?} product_id
         * @return {?}
         */
        WoocommerceProductsService.prototype.retrieveProductReviews = /**
         * Retrieve product reviews by product id
         * @param {?} product_id
         * @return {?}
         */
            function (product_id) {
                var _this = this;
                return this.httpClient.get("products/" + product_id + "/reviews")
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        /**
         * Retrieve product reviews by product id
         */
        /**
         * Retrieve product reviews by product id
         * @param {?} product_id
         * @param {?} review_id
         * @return {?}
         */
        WoocommerceProductsService.prototype.retrieveProductReview = /**
         * Retrieve product reviews by product id
         * @param {?} product_id
         * @param {?} review_id
         * @return {?}
         */
            function (product_id, review_id) {
                var _this = this;
                return this.httpClient.get("products/" + product_id + "/reviews/" + review_id)
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        WoocommerceProductsService.decorators = [
            { type: core.Injectable },
        ];
        /** @nocollapse */
        WoocommerceProductsService.ctorParameters = function () {
            return [
                { type: http.HttpClient, },
                { type: WoocommerceHelperService, },
            ];
        };
        return WoocommerceProductsService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    /** @enum {string} */
    var TagOrder = {
        "desc": 'desc',
        "asc": 'asc',
    };
    /** @enum {string} */
    var TagOrderBy = {
        "id": 'id',
        "include": 'include',
        "name": 'name',
        "slug": 'slug',
        "term_group": 'term_group',
        "description": 'description',
        "count": 'count',
    };

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var WoocommerceTagsService = (function () {
        function WoocommerceTagsService(httpClient, wooHelper) {
            this.httpClient = httpClient;
            this.wooHelper = wooHelper;
        }
        /**
         * Create a Tag
         * @param payload
         */
        /**
         * Create a Tag
         * @param {?} payload
         * @return {?}
         */
        WoocommerceTagsService.prototype.createTag = /**
         * Create a Tag
         * @param {?} payload
         * @return {?}
         */
            function (payload) {
                var _this = this;
                return this.httpClient.post("tags", payload)
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        /**
         * Retrieve a tag
         */
        /**
         * Retrieve a tag
         * @param {?} id
         * @return {?}
         */
        WoocommerceTagsService.prototype.retrieveTag = /**
         * Retrieve a tag
         * @param {?} id
         * @return {?}
         */
            function (id) {
                var _this = this;
                return this.httpClient.get("tags/" + id)
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        /**
         * Retrieve list of Tag
         */
        /**
         * Retrieve list of Tag
         * @param {?=} query
         * @return {?}
         */
        WoocommerceTagsService.prototype.retrieveTags = /**
         * Retrieve list of Tag
         * @param {?=} query
         * @return {?}
         */
            function (query) {
                var _this = this;
                return this.httpClient.get("tags", {
                    params: this.wooHelper.includeQuery(query)
                })
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        /**
         * Update Tag
         */
        /**
         * Update Tag
         * @param {?} id
         * @param {?} payload
         * @return {?}
         */
        WoocommerceTagsService.prototype.updateTag = /**
         * Update Tag
         * @param {?} id
         * @param {?} payload
         * @return {?}
         */
            function (id, payload) {
                var _this = this;
                return this.httpClient.put("tags/" + id, payload)
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        /**
         * Delete Tag
         */
        /**
         * Delete Tag
         * @param {?} id
         * @return {?}
         */
        WoocommerceTagsService.prototype.deleteagT = /**
         * Delete Tag
         * @param {?} id
         * @return {?}
         */
            function (id) {
                var _this = this;
                return this.httpClient.delete("tags/" + id)
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        WoocommerceTagsService.decorators = [
            { type: core.Injectable },
        ];
        /** @nocollapse */
        WoocommerceTagsService.ctorParameters = function () {
            return [
                { type: http.HttpClient, },
                { type: WoocommerceHelperService, },
            ];
        };
        return WoocommerceTagsService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    /** @enum {string} */
    var CategoryOrder = {
        "desc": 'desc',
        "asc": 'asc',
    };
    /** @enum {string} */
    var CategoryOrderBy = {
        "id": 'id',
        "include": 'include',
        "name": 'name',
        "slug": 'slug',
        "term_group": 'term_group',
        "description": 'description',
        "count": 'count',
    };

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var WoocommerceCategoriesService = (function () {
        function WoocommerceCategoriesService(httpClient, wooHelper) {
            this.httpClient = httpClient;
            this.wooHelper = wooHelper;
        }
        /**
         * Create a Category
         * @param payload: Product
         */
        /**
         * Create a Category
         * @param {?} category
         * @return {?}
         */
        WoocommerceCategoriesService.prototype.createCategory = /**
         * Create a Category
         * @param {?} category
         * @return {?}
         */
            function (category) {
                var _this = this;
                return this.httpClient.post("products/categories", category)
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        /**
         * Retrieve a Category
         */
        /**
         * Retrieve a Category
         * @param {?} id
         * @return {?}
         */
        WoocommerceCategoriesService.prototype.retrieveCategory = /**
         * Retrieve a Category
         * @param {?} id
         * @return {?}
         */
            function (id) {
                var _this = this;
                return this.httpClient.get("products/categories/" + id)
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        /**
         * Retrieve list of Category
         */
        /**
         * Retrieve list of Category
         * @param {?=} query
         * @return {?}
         */
        WoocommerceCategoriesService.prototype.retrieveCategories = /**
         * Retrieve list of Category
         * @param {?=} query
         * @return {?}
         */
            function (query) {
                var _this = this;
                return this.httpClient.get("products/categories", {
                    params: this.wooHelper.includeQuery(query)
                })
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        /**
         * Update Category
         */
        /**
         * Update Category
         * @param {?} id
         * @param {?} payload
         * @return {?}
         */
        WoocommerceCategoriesService.prototype.updateCategory = /**
         * Update Category
         * @param {?} id
         * @param {?} payload
         * @return {?}
         */
            function (id, payload) {
                var _this = this;
                return this.httpClient.put("products/categories/" + id, payload)
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        /**
         * Delete Category
         */
        /**
         * Delete Category
         * @param {?} id
         * @return {?}
         */
        WoocommerceCategoriesService.prototype.deleteCategory = /**
         * Delete Category
         * @param {?} id
         * @return {?}
         */
            function (id) {
                var _this = this;
                return this.httpClient.delete("products/categories/" + id)
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        WoocommerceCategoriesService.decorators = [
            { type: core.Injectable },
        ];
        /** @nocollapse */
        WoocommerceCategoriesService.ctorParameters = function () {
            return [
                { type: http.HttpClient, },
                { type: WoocommerceHelperService, },
            ];
        };
        return WoocommerceCategoriesService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    /** @enum {string} */
    var NonceMethod = {
        register: 'register',
        get_avatar: 'get_avatar',
        get_userinfo: 'get_userinfo',
        retrieve_password: 'retrieve_password',
        validate_auth_cookie: 'validate_auth_cookie',
        generate_auth_cookie: 'generate_auth_cookie',
        get_currentuserinfo: 'get_currentuserinfo',
        get_user_meta: 'get_user_meta',
        update_user_meta: 'update_user_meta',
        delete_user_meta: 'delete_user_meta',
        xprofile: 'xprofile',
        xprofile_update: 'xprofile_update',
        fb_connect: 'fb_connect',
    };

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var AuthService = (function () {
        function AuthService(httpClient, wooHelper) {
            this.httpClient = httpClient;
            this.wooHelper = wooHelper;
        }
        /**
         * @param {?} payload
         * @return {?}
         */
        AuthService.prototype.createNonce = /**
         * @param {?} payload
         * @return {?}
         */
            function (payload) {
                var _this = this;
                return this.httpClient.get("api/get_nonce/", { params: this.wooHelper.includeQuery(payload) })
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        /**
         * @param {?} registerData
         * @return {?}
         */
        AuthService.prototype.register = /**
         * @param {?} registerData
         * @return {?}
         */
            function (registerData) {
                var _this = this;
                var /** @type {?} */ payload = this.wooHelper.includeEncoded(registerData);
                return this.httpClient.post("api/user/register/", payload)
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        /**
         * @param {?} username
         * @return {?}
         */
        AuthService.prototype.retrivePassword = /**
         * @param {?} username
         * @return {?}
         */
            function (username) {
                var _this = this;
                var /** @type {?} */ payload = this.wooHelper.includeEncoded({ user_login: username });
                return this.httpClient.post("api/user/retrieve_password/", payload)
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        /**
         * @param {?} payload
         * @return {?}
         */
        AuthService.prototype.getAuthToken = /**
         * @param {?} payload
         * @return {?}
         */
            function (payload) {
                var _this = this;
                return this.httpClient.post("wp-json/jwt-auth/v1/token", payload)
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        /**
         * @param {?} data
         * @return {?}
         */
        AuthService.prototype.generateAuthCookie = /**
         * @param {?} data
         * @return {?}
         */
            function (data) {
                var _this = this;
                return this.httpClient.post("api/user/generate_auth_cookie/", this.wooHelper.includeEncoded(data))
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        AuthService.decorators = [
            { type: core.Injectable },
        ];
        /** @nocollapse */
        AuthService.ctorParameters = function () {
            return [
                { type: http.HttpClient, },
                { type: WoocommerceHelperService, },
            ];
        };
        return AuthService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var WoocommerceOrderService = (function () {
        function WoocommerceOrderService(httpClient, wooHelper) {
            this.httpClient = httpClient;
            this.wooHelper = wooHelper;
        }
        /**
         * @param {?} order
         * @return {?}
         */
        WoocommerceOrderService.prototype.createOrder = /**
         * @param {?} order
         * @return {?}
         */
            function (order) {
                var _this = this;
                return this.httpClient.post("orders", order)
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        /**
         * @param {?} id
         * @return {?}
         */
        WoocommerceOrderService.prototype.retriveOrder = /**
         * @param {?} id
         * @return {?}
         */
            function (id) {
                var _this = this;
                return this.httpClient.get("orders/" + id)
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        /**
         * @return {?}
         */
        WoocommerceOrderService.prototype.listAllOrder = /**
         * @return {?}
         */
            function () {
                var _this = this;
                return this.httpClient.get("orders")
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        /**
         * @param {?} order
         * @return {?}
         */
        WoocommerceOrderService.prototype.updateOrder = /**
         * @param {?} order
         * @return {?}
         */
            function (order) {
                var _this = this;
                return this.httpClient.put("orders/" + order.id, order)
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        /**
         * @param {?} id
         * @return {?}
         */
        WoocommerceOrderService.prototype.deleteOrder = /**
         * @param {?} id
         * @return {?}
         */
            function (id) {
                var _this = this;
                return this.httpClient.delete("orders/" + id)
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        WoocommerceOrderService.decorators = [
            { type: core.Injectable },
        ];
        /** @nocollapse */
        WoocommerceOrderService.ctorParameters = function () {
            return [
                { type: http.HttpClient, },
                { type: WoocommerceHelperService, },
            ];
        };
        return WoocommerceOrderService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var WoocommerceCouponService = (function () {
        function WoocommerceCouponService(httpClient, wooHelper) {
            this.httpClient = httpClient;
            this.wooHelper = wooHelper;
        }
        /**
         * @param {?} coupon
         * @return {?}
         */
        WoocommerceCouponService.prototype.createCoupon = /**
         * @param {?} coupon
         * @return {?}
         */
            function (coupon) {
                var _this = this;
                return this.httpClient.post("coupons", coupon)
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        /**
         * @param {?} id
         * @return {?}
         */
        WoocommerceCouponService.prototype.retriveCoupon = /**
         * @param {?} id
         * @return {?}
         */
            function (id) {
                var _this = this;
                return this.httpClient.get("coupons/" + id)
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        /**
         * @return {?}
         */
        WoocommerceCouponService.prototype.retriveCoupons = /**
         * @return {?}
         */
            function () {
                var _this = this;
                return this.httpClient.get("coupons")
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        /**
         * @param {?} coupon
         * @param {?} id
         * @return {?}
         */
        WoocommerceCouponService.prototype.updateCoupon = /**
         * @param {?} coupon
         * @param {?} id
         * @return {?}
         */
            function (coupon, id) {
                var _this = this;
                return this.httpClient.post("coupons/" + id, coupon)
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        /**
         * @param {?} id
         * @return {?}
         */
        WoocommerceCouponService.prototype.deleteCoupon = /**
         * @param {?} id
         * @return {?}
         */
            function (id) {
                var _this = this;
                return this.httpClient.delete("coupons/" + id + "?force=true")
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        WoocommerceCouponService.decorators = [
            { type: core.Injectable },
        ];
        /** @nocollapse */
        WoocommerceCouponService.ctorParameters = function () {
            return [
                { type: http.HttpClient, },
                { type: WoocommerceHelperService, },
            ];
        };
        return WoocommerceCouponService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var WoocommerceCustomerService = (function () {
        function WoocommerceCustomerService(httpClient, wooHelper) {
            this.httpClient = httpClient;
            this.wooHelper = wooHelper;
        }
        /**
         * Create a customer
         * @param customer: Customer
         */
        /**
         * Create a customer
         * @param {?} customer
         * @return {?}
         */
        WoocommerceCustomerService.prototype.createCustomers = /**
         * Create a customer
         * @param {?} customer
         * @return {?}
         */
            function (customer) {
                var _this = this;
                return this.httpClient.post("customers", customer)
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        /**
         * Retrive a customer
         * @param customer: Customer
         */
        /**
         * Retrive a customer
         * @param {?} id
         * @return {?}
         */
        WoocommerceCustomerService.prototype.retriveCustomers = /**
         * Retrive a customer
         * @param {?} id
         * @return {?}
         */
            function (id) {
                var _this = this;
                return this.httpClient.get("customers/" + id)
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        /**
         * Update a customer
         * @param customer: Customer
         */
        /**
         * Update a customer
         * @param {?} id
         * @param {?} customer
         * @return {?}
         */
        WoocommerceCustomerService.prototype.updateCustomers = /**
         * Update a customer
         * @param {?} id
         * @param {?} customer
         * @return {?}
         */
            function (id, customer) {
                var _this = this;
                return this.httpClient.put("customers/" + id, customer)
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        WoocommerceCustomerService.decorators = [
            { type: core.Injectable },
        ];
        /** @nocollapse */
        WoocommerceCustomerService.ctorParameters = function () {
            return [
                { type: http.HttpClient, },
                { type: WoocommerceHelperService, },
            ];
        };
        return WoocommerceCustomerService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */
    var WoocommerceOrderNoteService = (function () {
        function WoocommerceOrderNoteService(httpClient, wooHelper) {
            this.httpClient = httpClient;
            this.wooHelper = wooHelper;
        }
        /**
         * Create a order note
         * URL: https://woocommerce.github.io/woocommerce-rest-api-docs/#create-an-order-note
         */
        /**
         * Create a order note
         * URL: https://woocommerce.github.io/woocommerce-rest-api-docs/#create-an-order-note
         * @param {?} id
         * @param {?} note
         * @return {?}
         */
        WoocommerceOrderNoteService.prototype.createNote = /**
         * Create a order note
         * URL: https://woocommerce.github.io/woocommerce-rest-api-docs/#create-an-order-note
         * @param {?} id
         * @param {?} note
         * @return {?}
         */
            function (id, note) {
                var _this = this;
                return this.httpClient.post("orders/" + id + "/notes", note)
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        /**
         * Retrive a order note
         * URL: https://woocommerce.github.io/woocommerce-rest-api-docs/#retrieve-an-order-note
         */
        /**
         * Retrive a order note
         * URL: https://woocommerce.github.io/woocommerce-rest-api-docs/#retrieve-an-order-note
         * @param {?} id
         * @param {?} noteid
         * @return {?}
         */
        WoocommerceOrderNoteService.prototype.retriveNote = /**
         * Retrive a order note
         * URL: https://woocommerce.github.io/woocommerce-rest-api-docs/#retrieve-an-order-note
         * @param {?} id
         * @param {?} noteid
         * @return {?}
         */
            function (id, noteid) {
                var _this = this;
                return this.httpClient.get("orders/" + id + "/notes/" + noteid)
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        /**
        * List all order notes
        * URL: https://woocommerce.github.io/woocommerce-rest-api-docs/#list-all-order-notes
        */
        /**
         * List all order notes
         * URL: https://woocommerce.github.io/woocommerce-rest-api-docs/#list-all-order-notes
         * @param {?} id
         * @return {?}
         */
        WoocommerceOrderNoteService.prototype.retriveNotes = /**
         * List all order notes
         * URL: https://woocommerce.github.io/woocommerce-rest-api-docs/#list-all-order-notes
         * @param {?} id
         * @return {?}
         */
            function (id) {
                var _this = this;
                return this.httpClient.get("orders/" + id + "/notes/")
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        /**
         * Delete a order note
         * URL: https://woocommerce.github.io/woocommerce-rest-api-docs/#delete-an-order-note
         */
        /**
         * Delete a order note
         * URL: https://woocommerce.github.io/woocommerce-rest-api-docs/#delete-an-order-note
         * @param {?} id
         * @param {?} noteid
         * @return {?}
         */
        WoocommerceOrderNoteService.prototype.deleteNote = /**
         * Delete a order note
         * URL: https://woocommerce.github.io/woocommerce-rest-api-docs/#delete-an-order-note
         * @param {?} id
         * @param {?} noteid
         * @return {?}
         */
            function (id, noteid) {
                var _this = this;
                return this.httpClient.delete("orders/" + id + "/notes/" + noteid)
                    .pipe(operators.catchError(function (err) { return _this.wooHelper.handleError(err); }));
            };
        WoocommerceOrderNoteService.decorators = [
            { type: core.Injectable },
        ];
        /** @nocollapse */
        WoocommerceOrderNoteService.ctorParameters = function () {
            return [
                { type: http.HttpClient, },
                { type: WoocommerceHelperService, },
            ];
        };
        return WoocommerceOrderNoteService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes} checked by tsc
     */

    exports.WoocommerceHelperService = WoocommerceHelperService;
    exports.ProductOrder = ProductOrder;
    exports.ProductOrderBy = ProductOrderBy;
    exports.WoocommerceProductsService = WoocommerceProductsService;
    exports.TagOrder = TagOrder;
    exports.TagOrderBy = TagOrderBy;
    exports.WoocommerceTagsService = WoocommerceTagsService;
    exports.CategoryOrder = CategoryOrder;
    exports.CategoryOrderBy = CategoryOrderBy;
    exports.WoocommerceCategoriesService = WoocommerceCategoriesService;
    exports.NonceMethod = NonceMethod;
    exports.AuthService = AuthService;
    exports.WoocommerceOrderService = WoocommerceOrderService;
    exports.WoocommerceCouponService = WoocommerceCouponService;
    exports.WoocommerceCustomerService = WoocommerceCustomerService;
    exports.WoocommerceOrderNoteService = WoocommerceOrderNoteService;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
