// ngx-wooapi/fesm2015/ngx-wooapi.js
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { HttpClient } from '@angular/common/http';
import { catchError, map } from 'rxjs/operators';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class WoocommerceHelperService {
    constructor() { }
    /**
     * @param {?} error
     * @return {?}
     */
    handleError(error) {
        if (error.error instanceof ErrorEvent) {
            // A client-side or network error occurred. Handle it accordingly.
            console.error('An error occurred:', error.error);
        }
        else {
            // The backend returned an unsuccessful response code.
            // The response body may contain clues as to what went wrong,
            console.log('Backend response', error);
            return Observable.throw(error.error);
        }
        // return an observable with a user-facing error message
        return Observable.throw({ message: 'Something bad happened; please try again later.' });
    }
    /**
     * @param {?=} query
     * @return {?}
     */
    includeQuery(query = {}) {
        const /** @type {?} */ queryPatch = {};
        Object.keys(query).forEach(key => {
            queryPatch[key] = query[key].toString();
        });
        return queryPatch;
    }
    /**
     * @param {?=} query
     * @return {?}
     */
    includeEncoded(query = {}) {
        const /** @type {?} */ params = new FormData();
        Object.keys(query).forEach((key) => {
            params.append(key, query[key]);
        });
        console.log(params);
        return params;
    }
    /**
     * @param {?} response
     * @param {?=} responseBodyKey
     * @return {?}
     */
    includeResponseHeader(response, responseBodyKey) {
        const /** @type {?} */ headers = {};
        response.headers.keys().forEach(key => {
            headers[key] = response.headers.get(key);
        });
        let /** @type {?} */ responseData = {};
        if (responseBodyKey) {
            responseData[responseBodyKey] = response.body;
        }
        else {
            responseData = response.body;
        }
        responseData['headers'] = headers;
        return responseData;
    }
}
WoocommerceHelperService.decorators = [
    { type: Injectable },
];
/** @nocollapse */
WoocommerceHelperService.ctorParameters = () => [];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/** @enum {string} */
const ProductOrder = {
    "desc": 'desc',
    "asc": 'asc',
};
/** @enum {string} */
const ProductOrderBy = {
    "date": 'date',
    "id": 'id',
    "include": 'include',
    "title": 'title',
    "slug": 'slug',
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class WoocommerceProductsService {
    /**
     * @param {?} httpClient
     * @param {?} wooHelper
     */
    constructor(httpClient, wooHelper) {
        this.httpClient = httpClient;
        this.wooHelper = wooHelper;
    }
    /**
     * Create a Product
     * @param {?} payload
     * @return {?}
     */
    createProduct(payload) {
        return this.httpClient.post(`products`, payload)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Retrieve a product
     * @param {?} id
     * @return {?}
     */
    retrieveProduct(id) {
        return this.httpClient.get(`products/${id}`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Retrieve list of product
     * @param {?=} query
     * @return {?}
     */
    retrieveProducts(query) {
        return this.httpClient.get(`products`, { params: this.wooHelper.includeQuery(query), observe: 'response' })
            .pipe(map(value => this.wooHelper.includeResponseHeader(value, 'products')), catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Update Product
     * @param {?} id
     * @param {?} payload
     * @return {?}
     */
    updateProduct(id, payload) {
        return this.httpClient.put(`products/${id}`, payload)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Update Product
     * @param {?} id
     * @return {?}
     */
    deleteProduct(id) {
        return this.httpClient.delete(`products/${id}`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Retrieve product reviews by product id
     * @param {?} product_id
     * @return {?}
     */
    retrieveProductReviews(product_id) {
        return this.httpClient.get(`products/${product_id}/reviews`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Retrieve product reviews by product id
     * @param {?} product_id
     * @param {?} review_id
     * @return {?}
     */
    retrieveProductReview(product_id, review_id) {
        return this.httpClient.get(`products/${product_id}/reviews/${review_id}`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
}
WoocommerceProductsService.decorators = [
    { type: Injectable },
];
/** @nocollapse */
WoocommerceProductsService.ctorParameters = () => [
    { type: HttpClient, },
    { type: WoocommerceHelperService, },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/** @enum {string} */
const TagOrder = {
    "desc": 'desc',
    "asc": 'asc',
};
/** @enum {string} */
const TagOrderBy = {
    "id": 'id',
    "include": 'include',
    "name": 'name',
    "slug": 'slug',
    "term_group": 'term_group',
    "description": 'description',
    "count": 'count',
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class WoocommerceTagsService {
    /**
     * @param {?} httpClient
     * @param {?} wooHelper
     */
    constructor(httpClient, wooHelper) {
        this.httpClient = httpClient;
        this.wooHelper = wooHelper;
    }
    /**
     * Create a Tag
     * @param {?} payload
     * @return {?}
     */
    createTag(payload) {
        return this.httpClient.post(`tags`, payload)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Retrieve a tag
     * @param {?} id
     * @return {?}
     */
    retrieveTag(id) {
        return this.httpClient.get(`tags/${id}`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Retrieve list of Tag
     * @param {?=} query
     * @return {?}
     */
    retrieveTags(query) {
        return this.httpClient.get(`tags`, {
            params: this.wooHelper.includeQuery(query)
        })
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Update Tag
     * @param {?} id
     * @param {?} payload
     * @return {?}
     */
    updateTag(id, payload) {
        return this.httpClient.put(`tags/${id}`, payload)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Delete Tag
     * @param {?} id
     * @return {?}
     */
    deleteagT(id) {
        return this.httpClient.delete(`tags/${id}`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
}
WoocommerceTagsService.decorators = [
    { type: Injectable },
];
/** @nocollapse */
WoocommerceTagsService.ctorParameters = () => [
    { type: HttpClient, },
    { type: WoocommerceHelperService, },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/** @enum {string} */
const CategoryOrder = {
    "desc": 'desc',
    "asc": 'asc',
};
/** @enum {string} */
const CategoryOrderBy = {
    "id": 'id',
    "include": 'include',
    "name": 'name',
    "slug": 'slug',
    "term_group": 'term_group',
    "description": 'description',
    "count": 'count',
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class WoocommerceCategoriesService {
    /**
     * @param {?} httpClient
     * @param {?} wooHelper
     */
    constructor(httpClient, wooHelper) {
        this.httpClient = httpClient;
        this.wooHelper = wooHelper;
    }
    /**
     * Create a Category
     * @param {?} category
     * @return {?}
     */
    createCategory(category) {
        return this.httpClient.post(`products/categories`, category)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Retrieve a Category
     * @param {?} id
     * @return {?}
     */
    retrieveCategory(id) {
        return this.httpClient.get(`products/categories/${id}`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Retrieve list of Category
     * @param {?=} query
     * @return {?}
     */
    retrieveCategories(query) {
        return this.httpClient.get(`products/categories`, {
            params: this.wooHelper.includeQuery(query)
        })
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Update Category
     * @param {?} id
     * @param {?} payload
     * @return {?}
     */
    updateCategory(id, payload) {
        return this.httpClient.put(`products/categories/${id}`, payload)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Delete Category
     * @param {?} id
     * @return {?}
     */
    deleteCategory(id) {
        return this.httpClient.delete(`products/categories/${id}`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
}
WoocommerceCategoriesService.decorators = [
    { type: Injectable },
];
/** @nocollapse */
WoocommerceCategoriesService.ctorParameters = () => [
    { type: HttpClient, },
    { type: WoocommerceHelperService, },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/** @enum {string} */
const NonceMethod = {
    register: 'register',
    get_avatar: 'get_avatar',
    get_userinfo: 'get_userinfo',
    retrieve_password: 'retrieve_password',
    validate_auth_cookie: 'validate_auth_cookie',
    generate_auth_cookie: 'generate_auth_cookie',
    get_currentuserinfo: 'get_currentuserinfo',
    get_user_meta: 'get_user_meta',
    update_user_meta: 'update_user_meta',
    delete_user_meta: 'delete_user_meta',
    xprofile: 'xprofile',
    xprofile_update: 'xprofile_update',
    fb_connect: 'fb_connect',
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class AuthService {
    /**
     * @param {?} httpClient
     * @param {?} wooHelper
     */
    constructor(httpClient, wooHelper) {
        this.httpClient = httpClient;
        this.wooHelper = wooHelper;
    }
    /**
     * @param {?} payload
     * @return {?}
     */
    createNonce(payload) {
        return this.httpClient.get(`api/get_nonce/`, { params: this.wooHelper.includeQuery(payload) })
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * @param {?} registerData
     * @return {?}
     */
    register(registerData) {
        const /** @type {?} */ payload = this.wooHelper.includeEncoded(registerData);
        return this.httpClient.post(`api/user/register/`, payload)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * @param {?} username
     * @return {?}
     */
    retrivePassword(username) {
        const /** @type {?} */ payload = this.wooHelper.includeEncoded({ user_login: username });
        return this.httpClient.post(`api/user/retrieve_password/`, payload)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * @param {?} payload
     * @return {?}
     */
    getAuthToken(payload) {
        return this.httpClient.post(`wp-json/jwt-auth/v1/token`, payload)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * @param {?} data
     * @return {?}
     */
    generateAuthCookie(data) {
        return this.httpClient.post(`api/user/generate_auth_cookie/`, this.wooHelper.includeEncoded(data))
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
}
AuthService.decorators = [
    { type: Injectable },
];
/** @nocollapse */
AuthService.ctorParameters = () => [
    { type: HttpClient, },
    { type: WoocommerceHelperService, },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class WoocommerceOrderService {
    /**
     * @param {?} httpClient
     * @param {?} wooHelper
     */
    constructor(httpClient, wooHelper) {
        this.httpClient = httpClient;
        this.wooHelper = wooHelper;
    }
    /**
     * @param {?} order
     * @return {?}
     */
    createOrder(order) {
        return this.httpClient.post(`orders`, order)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * @param {?} id
     * @return {?}
     */
    retriveOrder(id) {
        return this.httpClient.get(`orders/${id}`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * @return {?}
     */
    listAllOrder() {
        return this.httpClient.get(`orders`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * @param {?} order
     * @return {?}
     */
    updateOrder(order) {
        return this.httpClient.put(`orders/${order.id}`, order)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * @param {?} id
     * @return {?}
     */
    deleteOrder(id) {
        return this.httpClient.delete(`orders/${id}`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
}
WoocommerceOrderService.decorators = [
    { type: Injectable },
];
/** @nocollapse */
WoocommerceOrderService.ctorParameters = () => [
    { type: HttpClient, },
    { type: WoocommerceHelperService, },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class WoocommerceCouponService {
    /**
     * @param {?} httpClient
     * @param {?} wooHelper
     */
    constructor(httpClient, wooHelper) {
        this.httpClient = httpClient;
        this.wooHelper = wooHelper;
    }
    /**
     * @param {?} coupon
     * @return {?}
     */
    createCoupon(coupon) {
        return this.httpClient.post(`coupons`, coupon)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * @param {?} id
     * @return {?}
     */
    retriveCoupon(id) {
        return this.httpClient.get(`coupons/${id}`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * @return {?}
     */
    retriveCoupons() {
        return this.httpClient.get(`coupons`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * @param {?} coupon
     * @param {?} id
     * @return {?}
     */
    updateCoupon(coupon, id) {
        return this.httpClient.post(`coupons/${id}`, coupon)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * @param {?} id
     * @return {?}
     */
    deleteCoupon(id) {
        return this.httpClient.delete(`coupons/${id}?force=true`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
}
WoocommerceCouponService.decorators = [
    { type: Injectable },
];
/** @nocollapse */
WoocommerceCouponService.ctorParameters = () => [
    { type: HttpClient, },
    { type: WoocommerceHelperService, },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class WoocommerceCustomerService {
    /**
     * @param {?} httpClient
     * @param {?} wooHelper
     */
    constructor(httpClient, wooHelper) {
        this.httpClient = httpClient;
        this.wooHelper = wooHelper;
    }
    /**
     * Create a customer
     * @param {?} customer
     * @return {?}
     */
    createCustomers(customer) {
        return this.httpClient.post(`customers`, customer)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Retrive a customer
     * @param {?} id
     * @return {?}
     */
    retriveCustomers(id) {
        return this.httpClient.get(`customers/${id}`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Update a customer
     * @param {?} id
     * @param {?} customer
     * @return {?}
     */
    updateCustomers(id, customer) {
        return this.httpClient.put(`customers/${id}`, customer)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
}
WoocommerceCustomerService.decorators = [
    { type: Injectable },
];
/** @nocollapse */
WoocommerceCustomerService.ctorParameters = () => [
    { type: HttpClient, },
    { type: WoocommerceHelperService, },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class WoocommerceOrderNoteService {
    /**
     * @param {?} httpClient
     * @param {?} wooHelper
     */
    constructor(httpClient, wooHelper) {
        this.httpClient = httpClient;
        this.wooHelper = wooHelper;
    }
    /**
     * Create a order note
     * URL: https://woocommerce.github.io/woocommerce-rest-api-docs/#create-an-order-note
     * @param {?} id
     * @param {?} note
     * @return {?}
     */
    createNote(id, note) {
        return this.httpClient.post(`orders/${id}/notes`, note)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Retrive a order note
     * URL: https://woocommerce.github.io/woocommerce-rest-api-docs/#retrieve-an-order-note
     * @param {?} id
     * @param {?} noteid
     * @return {?}
     */
    retriveNote(id, noteid) {
        return this.httpClient.get(`orders/${id}/notes/${noteid}`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * List all order notes
     * URL: https://woocommerce.github.io/woocommerce-rest-api-docs/#list-all-order-notes
     * @param {?} id
     * @return {?}
     */
    retriveNotes(id) {
        return this.httpClient.get(`orders/${id}/notes/`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Delete a order note
     * URL: https://woocommerce.github.io/woocommerce-rest-api-docs/#delete-an-order-note
     * @param {?} id
     * @param {?} noteid
     * @return {?}
     */
    deleteNote(id, noteid) {
        return this.httpClient.delete(`orders/${id}/notes/${noteid}`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
}
WoocommerceOrderNoteService.decorators = [
    { type: Injectable },
];
/** @nocollapse */
WoocommerceOrderNoteService.ctorParameters = () => [
    { type: HttpClient, },
    { type: WoocommerceHelperService, },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

export { WoocommerceHelperService, ProductOrder, ProductOrderBy, WoocommerceProductsService, TagOrder, TagOrderBy, WoocommerceTagsService, CategoryOrder, CategoryOrderBy, WoocommerceCategoriesService, NonceMethod, AuthService, WoocommerceOrderService, WoocommerceCouponService, WoocommerceCustomerService, WoocommerceOrderNoteService };
