// ngx-wooapi/esm5/tags/woocommerce-tags.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { catchError } from 'rxjs/operators';
import { WoocommerceHelperService } from '../helper.service';
var WoocommerceTagsService = /** @class */ (function () {
    function WoocommerceTagsService(httpClient, wooHelper) {
        this.httpClient = httpClient;
        this.wooHelper = wooHelper;
    }
    /**
     * Create a Tag
     * @param payload
     */
    /**
     * Create a Tag
     * @param {?} payload
     * @return {?}
     */
    WoocommerceTagsService.prototype.createTag = /**
     * Create a Tag
     * @param {?} payload
     * @return {?}
     */
    function (payload) {
        var _this = this;
        return this.httpClient.post("tags", payload)
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    /**
     * Retrieve a tag
     */
    /**
     * Retrieve a tag
     * @param {?} id
     * @return {?}
     */
    WoocommerceTagsService.prototype.retrieveTag = /**
     * Retrieve a tag
     * @param {?} id
     * @return {?}
     */
    function (id) {
        var _this = this;
        return this.httpClient.get("tags/" + id)
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    /**
     * Retrieve list of Tag
     */
    /**
     * Retrieve list of Tag
     * @param {?=} query
     * @return {?}
     */
    WoocommerceTagsService.prototype.retrieveTags = /**
     * Retrieve list of Tag
     * @param {?=} query
     * @return {?}
     */
    function (query) {
        var _this = this;
        return this.httpClient.get("tags", {
            params: this.wooHelper.includeQuery(query)
        })
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    /**
     * Update Tag
     */
    /**
     * Update Tag
     * @param {?} id
     * @param {?} payload
     * @return {?}
     */
    WoocommerceTagsService.prototype.updateTag = /**
     * Update Tag
     * @param {?} id
     * @param {?} payload
     * @return {?}
     */
    function (id, payload) {
        var _this = this;
        return this.httpClient.put("tags/" + id, payload)
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    /**
     * Delete Tag
     */
    /**
     * Delete Tag
     * @param {?} id
     * @return {?}
     */
    WoocommerceTagsService.prototype.deleteagT = /**
     * Delete Tag
     * @param {?} id
     * @return {?}
     */
    function (id) {
        var _this = this;
        return this.httpClient.delete("tags/" + id)
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    WoocommerceTagsService.decorators = [
        { type: Injectable },
    ];
    /** @nocollapse */
    WoocommerceTagsService.ctorParameters = function () { return [
        { type: HttpClient, },
        { type: WoocommerceHelperService, },
    ]; };
    return WoocommerceTagsService;
}());
export { WoocommerceTagsService };
function WoocommerceTagsService_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    WoocommerceTagsService.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    WoocommerceTagsService.ctorParameters;
    /** @type {?} */
    WoocommerceTagsService.prototype.httpClient;
    /** @type {?} */
    WoocommerceTagsService.prototype.wooHelper;
}
