// ngx-wooapi/esm5/coupon/coupon.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { catchError } from 'rxjs/operators';
import { WoocommerceHelperService } from '../helper.service';
var WoocommerceCouponService = /** @class */ (function () {
    function WoocommerceCouponService(httpClient, wooHelper) {
        this.httpClient = httpClient;
        this.wooHelper = wooHelper;
    }
    /**
     * @param {?} coupon
     * @return {?}
     */
    WoocommerceCouponService.prototype.createCoupon = /**
     * @param {?} coupon
     * @return {?}
     */
    function (coupon) {
        var _this = this;
        return this.httpClient.post("coupons", coupon)
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    /**
     * @param {?} id
     * @return {?}
     */
    WoocommerceCouponService.prototype.retriveCoupon = /**
     * @param {?} id
     * @return {?}
     */
    function (id) {
        var _this = this;
        return this.httpClient.get("coupons/" + id)
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    /**
     * @return {?}
     */
    WoocommerceCouponService.prototype.retriveCoupons = /**
     * @return {?}
     */
    function () {
        var _this = this;
        return this.httpClient.get("coupons")
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    /**
     * @param {?} coupon
     * @param {?} id
     * @return {?}
     */
    WoocommerceCouponService.prototype.updateCoupon = /**
     * @param {?} coupon
     * @param {?} id
     * @return {?}
     */
    function (coupon, id) {
        var _this = this;
        return this.httpClient.post("coupons/" + id, coupon)
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    /**
     * @param {?} id
     * @return {?}
     */
    WoocommerceCouponService.prototype.deleteCoupon = /**
     * @param {?} id
     * @return {?}
     */
    function (id) {
        var _this = this;
        return this.httpClient.delete("coupons/" + id + "?force=true")
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    WoocommerceCouponService.decorators = [
        { type: Injectable },
    ];
    /** @nocollapse */
    WoocommerceCouponService.ctorParameters = function () { return [
        { type: HttpClient, },
        { type: WoocommerceHelperService, },
    ]; };
    return WoocommerceCouponService;
}());
export { WoocommerceCouponService };
function WoocommerceCouponService_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    WoocommerceCouponService.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    WoocommerceCouponService.ctorParameters;
    /** @type {?} */
    WoocommerceCouponService.prototype.httpClient;
    /** @type {?} */
    WoocommerceCouponService.prototype.wooHelper;
}
