// ngx-wooapi/esm2015/order-notes/order-notes.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { catchError } from 'rxjs/operators';
import { WoocommerceHelperService } from '../helper.service';
export class WoocommerceOrderNoteService {
    /**
     * @param {?} httpClient
     * @param {?} wooHelper
     */
    constructor(httpClient, wooHelper) {
        this.httpClient = httpClient;
        this.wooHelper = wooHelper;
    }
    /**
     * Create a order note
     * URL: https://woocommerce.github.io/woocommerce-rest-api-docs/#create-an-order-note
     * @param {?} id
     * @param {?} note
     * @return {?}
     */
    createNote(id, note) {
        return this.httpClient.post(`orders/${id}/notes`, note)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Retrive a order note
     * URL: https://woocommerce.github.io/woocommerce-rest-api-docs/#retrieve-an-order-note
     * @param {?} id
     * @param {?} noteid
     * @return {?}
     */
    retriveNote(id, noteid) {
        return this.httpClient.get(`orders/${id}/notes/${noteid}`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * List all order notes
     * URL: https://woocommerce.github.io/woocommerce-rest-api-docs/#list-all-order-notes
     * @param {?} id
     * @return {?}
     */
    retriveNotes(id) {
        return this.httpClient.get(`orders/${id}/notes/`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Delete a order note
     * URL: https://woocommerce.github.io/woocommerce-rest-api-docs/#delete-an-order-note
     * @param {?} id
     * @param {?} noteid
     * @return {?}
     */
    deleteNote(id, noteid) {
        return this.httpClient.delete(`orders/${id}/notes/${noteid}`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
}
WoocommerceOrderNoteService.decorators = [
    { type: Injectable },
];
/** @nocollapse */
WoocommerceOrderNoteService.ctorParameters = () => [
    { type: HttpClient, },
    { type: WoocommerceHelperService, },
];
function WoocommerceOrderNoteService_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    WoocommerceOrderNoteService.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    WoocommerceOrderNoteService.ctorParameters;
    /** @type {?} */
    WoocommerceOrderNoteService.prototype.httpClient;
    /** @type {?} */
    WoocommerceOrderNoteService.prototype.wooHelper;
}
