// ngx-wooapi/esm2015/tags/product-tags.interface.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * @record
 */
export function ProductTag() { }
function ProductTag_tsickle_Closure_declarations() {
    /** @type {?|undefined} */
    ProductTag.prototype.id;
    /** @type {?|undefined} */
    ProductTag.prototype.name;
    /** @type {?|undefined} */
    ProductTag.prototype.slug;
    /** @type {?|undefined} */
    ProductTag.prototype.description;
    /** @type {?|undefined} */
    ProductTag.prototype.count;
}
/** @enum {string} */
const TagOrder = {
    "desc": 'desc',
    "asc": 'asc',
};
export { TagOrder };
/** @enum {string} */
const TagOrderBy = {
    "id": 'id',
    "include": 'include',
    "name": 'name',
    "slug": 'slug',
    "term_group": 'term_group',
    "description": 'description',
    "count": 'count',
};
export { TagOrderBy };
/**
 * @record
 */
export function TagQuery() { }
function TagQuery_tsickle_Closure_declarations() {
    /** @type {?|undefined} */
    TagQuery.prototype.context;
    /** @type {?|undefined} */
    TagQuery.prototype.page;
    /** @type {?|undefined} */
    TagQuery.prototype.per_page;
    /** @type {?|undefined} */
    TagQuery.prototype.search;
    /** @type {?|undefined} */
    TagQuery.prototype.exclude;
    /** @type {?|undefined} */
    TagQuery.prototype.include;
    /** @type {?} */
    TagQuery.prototype.offset;
    /** @type {?|undefined} */
    TagQuery.prototype.order;
    /** @type {?|undefined} */
    TagQuery.prototype.orderby;
    /** @type {?} */
    TagQuery.prototype.hide_empty;
    /** @type {?} */
    TagQuery.prototype.product;
    /** @type {?} */
    TagQuery.prototype.slug;
}
