// ngx-wooapi/esm2015/tags/woocommerce-tags.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { catchError } from 'rxjs/operators';
import { WoocommerceHelperService } from '../helper.service';
export class WoocommerceTagsService {
    /**
     * @param {?} httpClient
     * @param {?} wooHelper
     */
    constructor(httpClient, wooHelper) {
        this.httpClient = httpClient;
        this.wooHelper = wooHelper;
    }
    /**
     * Create a Tag
     * @param {?} payload
     * @return {?}
     */
    createTag(payload) {
        return this.httpClient.post(`tags`, payload)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Retrieve a tag
     * @param {?} id
     * @return {?}
     */
    retrieveTag(id) {
        return this.httpClient.get(`tags/${id}`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Retrieve list of Tag
     * @param {?=} query
     * @return {?}
     */
    retrieveTags(query) {
        return this.httpClient.get(`tags`, {
            params: this.wooHelper.includeQuery(query)
        })
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Update Tag
     * @param {?} id
     * @param {?} payload
     * @return {?}
     */
    updateTag(id, payload) {
        return this.httpClient.put(`tags/${id}`, payload)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Delete Tag
     * @param {?} id
     * @return {?}
     */
    deleteagT(id) {
        return this.httpClient.delete(`tags/${id}`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
}
WoocommerceTagsService.decorators = [
    { type: Injectable },
];
/** @nocollapse */
WoocommerceTagsService.ctorParameters = () => [
    { type: HttpClient, },
    { type: WoocommerceHelperService, },
];
function WoocommerceTagsService_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    WoocommerceTagsService.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    WoocommerceTagsService.ctorParameters;
    /** @type {?} */
    WoocommerceTagsService.prototype.httpClient;
    /** @type {?} */
    WoocommerceTagsService.prototype.wooHelper;
}
