// ngx-wooapi/esm5/auth/auth.interface.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/** @enum {string} */
var NonceMethod = {
    register: 'register',
    get_avatar: 'get_avatar',
    get_userinfo: 'get_userinfo',
    retrieve_password: 'retrieve_password',
    validate_auth_cookie: 'validate_auth_cookie',
    generate_auth_cookie: 'generate_auth_cookie',
    get_currentuserinfo: 'get_currentuserinfo',
    get_user_meta: 'get_user_meta',
    update_user_meta: 'update_user_meta',
    delete_user_meta: 'delete_user_meta',
    xprofile: 'xprofile',
    xprofile_update: 'xprofile_update',
    fb_connect: 'fb_connect',
};
export { NonceMethod };
/**
 * @record
 */
export function CreateNonce() { }
function CreateNonce_tsickle_Closure_declarations() {
    /** @type {?} */
    CreateNonce.prototype.controller;
    /** @type {?} */
    CreateNonce.prototype.method;
}
/**
 * @record
 */
export function CreateNonceRes() { }
function CreateNonceRes_tsickle_Closure_declarations() {
    /** @type {?} */
    CreateNonceRes.prototype.status;
    /** @type {?} */
    CreateNonceRes.prototype.controller;
    /** @type {?} */
    CreateNonceRes.prototype.method;
    /** @type {?} */
    CreateNonceRes.prototype.nonce;
}
/**
 * @record
 */
export function RegisterPayload() { }
function RegisterPayload_tsickle_Closure_declarations() {
    /** @type {?} */
    RegisterPayload.prototype.username;
    /** @type {?} */
    RegisterPayload.prototype.email;
    /** @type {?|undefined} */
    RegisterPayload.prototype.user_pass;
    /** @type {?} */
    RegisterPayload.prototype.nonce;
    /** @type {?} */
    RegisterPayload.prototype.display_name;
    /** @type {?} */
    RegisterPayload.prototype.notify;
}
/**
 * @record
 */
export function RegisterRsponse() { }
function RegisterRsponse_tsickle_Closure_declarations() {
    /** @type {?} */
    RegisterRsponse.prototype.cookie;
    /** @type {?} */
    RegisterRsponse.prototype.status;
    /** @type {?} */
    RegisterRsponse.prototype.user_id;
}
/**
 * @record
 */
export function LoginPayload() { }
function LoginPayload_tsickle_Closure_declarations() {
    /** @type {?} */
    LoginPayload.prototype.username;
    /** @type {?} */
    LoginPayload.prototype.password;
}
