// ngx-wooapi/esm5/auth/auth.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { catchError } from 'rxjs/operators';
import { WoocommerceHelperService } from '../helper.service';
var AuthService = /** @class */ (function () {
    function AuthService(httpClient, wooHelper) {
        this.httpClient = httpClient;
        this.wooHelper = wooHelper;
    }
    /**
     * @param {?} payload
     * @return {?}
     */
    AuthService.prototype.createNonce = /**
     * @param {?} payload
     * @return {?}
     */
    function (payload) {
        var _this = this;
        return this.httpClient.get("api/get_nonce/", { params: this.wooHelper.includeQuery(payload) })
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    /**
     * @param {?} registerData
     * @return {?}
     */
    AuthService.prototype.register = /**
     * @param {?} registerData
     * @return {?}
     */
    function (registerData) {
        var _this = this;
        var /** @type {?} */ payload = this.wooHelper.includeEncoded(registerData);
        return this.httpClient.post("api/user/register/", payload)
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    /**
     * @param {?} username
     * @return {?}
     */
    AuthService.prototype.retrivePassword = /**
     * @param {?} username
     * @return {?}
     */
    function (username) {
        var _this = this;
        var /** @type {?} */ payload = this.wooHelper.includeEncoded({ user_login: username });
        return this.httpClient.post("api/user/retrieve_password/", payload)
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    /**
     * @param {?} payload
     * @return {?}
     */
    AuthService.prototype.getAuthToken = /**
     * @param {?} payload
     * @return {?}
     */
    function (payload) {
        var _this = this;
        return this.httpClient.post("wp-json/jwt-auth/v1/token", payload)
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    /**
     * @param {?} data
     * @return {?}
     */
    AuthService.prototype.generateAuthCookie = /**
     * @param {?} data
     * @return {?}
     */
    function (data) {
        var _this = this;
        return this.httpClient.post("api/user/generate_auth_cookie/", this.wooHelper.includeEncoded(data))
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    AuthService.decorators = [
        { type: Injectable },
    ];
    /** @nocollapse */
    AuthService.ctorParameters = function () { return [
        { type: HttpClient, },
        { type: WoocommerceHelperService, },
    ]; };
    return AuthService;
}());
export { AuthService };
function AuthService_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    AuthService.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    AuthService.ctorParameters;
    /** @type {?} */
    AuthService.prototype.httpClient;
    /** @type {?} */
    AuthService.prototype.wooHelper;
}
