// ngx-wooapi/esm5/customer/customer.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { catchError } from 'rxjs/operators';
import { WoocommerceHelperService } from '../helper.service';
var WoocommerceCustomerService = /** @class */ (function () {
    function WoocommerceCustomerService(httpClient, wooHelper) {
        this.httpClient = httpClient;
        this.wooHelper = wooHelper;
    }
    /**
     * Create a customer
     * @param customer: Customer
     */
    /**
     * Create a customer
     * @param {?} customer
     * @return {?}
     */
    WoocommerceCustomerService.prototype.createCustomers = /**
     * Create a customer
     * @param {?} customer
     * @return {?}
     */
    function (customer) {
        var _this = this;
        return this.httpClient.post("customers", customer)
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    /**
     * Retrive a customer
     * @param customer: Customer
     */
    /**
     * Retrive a customer
     * @param {?} id
     * @return {?}
     */
    WoocommerceCustomerService.prototype.retriveCustomers = /**
     * Retrive a customer
     * @param {?} id
     * @return {?}
     */
    function (id) {
        var _this = this;
        return this.httpClient.get("customers/" + id)
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    /**
     * Update a customer
     * @param customer: Customer
     */
    /**
     * Update a customer
     * @param {?} id
     * @param {?} customer
     * @return {?}
     */
    WoocommerceCustomerService.prototype.updateCustomers = /**
     * Update a customer
     * @param {?} id
     * @param {?} customer
     * @return {?}
     */
    function (id, customer) {
        var _this = this;
        return this.httpClient.put("customers/" + id, customer)
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    WoocommerceCustomerService.decorators = [
        { type: Injectable },
    ];
    /** @nocollapse */
    WoocommerceCustomerService.ctorParameters = function () { return [
        { type: HttpClient, },
        { type: WoocommerceHelperService, },
    ]; };
    return WoocommerceCustomerService;
}());
export { WoocommerceCustomerService };
function WoocommerceCustomerService_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    WoocommerceCustomerService.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    WoocommerceCustomerService.ctorParameters;
    /** @type {?} */
    WoocommerceCustomerService.prototype.httpClient;
    /** @type {?} */
    WoocommerceCustomerService.prototype.wooHelper;
}
