// ngx-wooapi/esm2015/auth/auth.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { catchError } from 'rxjs/operators';
import { WoocommerceHelperService } from '../helper.service';
export class AuthService {
    /**
     * @param {?} httpClient
     * @param {?} wooHelper
     */
    constructor(httpClient, wooHelper) {
        this.httpClient = httpClient;
        this.wooHelper = wooHelper;
    }
    /**
     * @param {?} payload
     * @return {?}
     */
    createNonce(payload) {
        return this.httpClient.get(`api/get_nonce/`, { params: this.wooHelper.includeQuery(payload) })
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * @param {?} registerData
     * @return {?}
     */
    register(registerData) {
        const /** @type {?} */ payload = this.wooHelper.includeEncoded(registerData);
        return this.httpClient.post(`api/user/register/`, payload)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * @param {?} username
     * @return {?}
     */
    retrivePassword(username) {
        const /** @type {?} */ payload = this.wooHelper.includeEncoded({ user_login: username });
        return this.httpClient.post(`api/user/retrieve_password/`, payload)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * @param {?} payload
     * @return {?}
     */
    getAuthToken(payload) {
        return this.httpClient.post(`wp-json/jwt-auth/v1/token`, payload)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * @param {?} data
     * @return {?}
     */
    generateAuthCookie(data) {
        return this.httpClient.post(`api/user/generate_auth_cookie/`, this.wooHelper.includeEncoded(data))
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
}
AuthService.decorators = [
    { type: Injectable },
];
/** @nocollapse */
AuthService.ctorParameters = () => [
    { type: HttpClient, },
    { type: WoocommerceHelperService, },
];
function AuthService_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    AuthService.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    AuthService.ctorParameters;
    /** @type {?} */
    AuthService.prototype.httpClient;
    /** @type {?} */
    AuthService.prototype.wooHelper;
}
