// ngx-wooapi/esm2015/coupon/coupon.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { catchError } from 'rxjs/operators';
import { WoocommerceHelperService } from '../helper.service';
export class WoocommerceCouponService {
    /**
     * @param {?} httpClient
     * @param {?} wooHelper
     */
    constructor(httpClient, wooHelper) {
        this.httpClient = httpClient;
        this.wooHelper = wooHelper;
    }
    /**
     * @param {?} coupon
     * @return {?}
     */
    createCoupon(coupon) {
        return this.httpClient.post(`coupons`, coupon)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * @param {?} id
     * @return {?}
     */
    retriveCoupon(id) {
        return this.httpClient.get(`coupons/${id}`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * @return {?}
     */
    retriveCoupons() {
        return this.httpClient.get(`coupons`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * @param {?} coupon
     * @param {?} id
     * @return {?}
     */
    updateCoupon(coupon, id) {
        return this.httpClient.post(`coupons/${id}`, coupon)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * @param {?} id
     * @return {?}
     */
    deleteCoupon(id) {
        return this.httpClient.delete(`coupons/${id}?force=true`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
}
WoocommerceCouponService.decorators = [
    { type: Injectable },
];
/** @nocollapse */
WoocommerceCouponService.ctorParameters = () => [
    { type: HttpClient, },
    { type: WoocommerceHelperService, },
];
function WoocommerceCouponService_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    WoocommerceCouponService.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    WoocommerceCouponService.ctorParameters;
    /** @type {?} */
    WoocommerceCouponService.prototype.httpClient;
    /** @type {?} */
    WoocommerceCouponService.prototype.wooHelper;
}
