// ngx-wooapi/esm5/products/woocommerce-products.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { catchError, map } from 'rxjs/operators';
import { WoocommerceHelperService } from '../helper.service';
var WoocommerceProductsService = /** @class */ (function () {
    function WoocommerceProductsService(httpClient, wooHelper) {
        this.httpClient = httpClient;
        this.wooHelper = wooHelper;
    }
    /**
     * Create a Product
     * @param payload: Product
     */
    /**
     * Create a Product
     * @param {?} payload
     * @return {?}
     */
    WoocommerceProductsService.prototype.createProduct = /**
     * Create a Product
     * @param {?} payload
     * @return {?}
     */
    function (payload) {
        var _this = this;
        return this.httpClient.post("products", payload)
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    /**
     * Retrieve a product
     */
    /**
     * Retrieve a product
     * @param {?} id
     * @return {?}
     */
    WoocommerceProductsService.prototype.retrieveProduct = /**
     * Retrieve a product
     * @param {?} id
     * @return {?}
     */
    function (id) {
        var _this = this;
        return this.httpClient.get("products/" + id)
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    /**
     * Retrieve list of product
     */
    /**
     * Retrieve list of product
     * @param {?=} query
     * @return {?}
     */
    WoocommerceProductsService.prototype.retrieveProducts = /**
     * Retrieve list of product
     * @param {?=} query
     * @return {?}
     */
    function (query) {
        var _this = this;
        return this.httpClient.get("products", { params: this.wooHelper.includeQuery(query), observe: 'response' })
            .pipe(map(function (value) { return _this.wooHelper.includeResponseHeader(value, 'products'); }), catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    /**
     * Update Product
     */
    /**
     * Update Product
     * @param {?} id
     * @param {?} payload
     * @return {?}
     */
    WoocommerceProductsService.prototype.updateProduct = /**
     * Update Product
     * @param {?} id
     * @param {?} payload
     * @return {?}
     */
    function (id, payload) {
        var _this = this;
        return this.httpClient.put("products/" + id, payload)
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    /**
     * Update Product
     */
    /**
     * Update Product
     * @param {?} id
     * @return {?}
     */
    WoocommerceProductsService.prototype.deleteProduct = /**
     * Update Product
     * @param {?} id
     * @return {?}
     */
    function (id) {
        var _this = this;
        return this.httpClient.delete("products/" + id)
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    /**
     * Retrieve product reviews by product id
     */
    /**
     * Retrieve product reviews by product id
     * @param {?} product_id
     * @return {?}
     */
    WoocommerceProductsService.prototype.retrieveProductReviews = /**
     * Retrieve product reviews by product id
     * @param {?} product_id
     * @return {?}
     */
    function (product_id) {
        var _this = this;
        return this.httpClient.get("products/" + product_id + "/reviews")
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    /**
     * Retrieve product reviews by product id
     */
    /**
     * Retrieve product reviews by product id
     * @param {?} product_id
     * @param {?} review_id
     * @return {?}
     */
    WoocommerceProductsService.prototype.retrieveProductReview = /**
     * Retrieve product reviews by product id
     * @param {?} product_id
     * @param {?} review_id
     * @return {?}
     */
    function (product_id, review_id) {
        var _this = this;
        return this.httpClient.get("products/" + product_id + "/reviews/" + review_id)
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    WoocommerceProductsService.decorators = [
        { type: Injectable },
    ];
    /** @nocollapse */
    WoocommerceProductsService.ctorParameters = function () { return [
        { type: HttpClient, },
        { type: WoocommerceHelperService, },
    ]; };
    return WoocommerceProductsService;
}());
export { WoocommerceProductsService };
function WoocommerceProductsService_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    WoocommerceProductsService.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    WoocommerceProductsService.ctorParameters;
    /** @type {?} */
    WoocommerceProductsService.prototype.httpClient;
    /** @type {?} */
    WoocommerceProductsService.prototype.wooHelper;
}
