// ngx-wooapi/esm2015/helper.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
export class WoocommerceHelperService {
    constructor() { }
    /**
     * @param {?} error
     * @return {?}
     */
    handleError(error) {
        if (error.error instanceof ErrorEvent) {
            // A client-side or network error occurred. Handle it accordingly.
            console.error('An error occurred:', error.error);
        }
        else {
            // The backend returned an unsuccessful response code.
            // The response body may contain clues as to what went wrong,
            console.log('Backend response', error);
            return Observable.throw(error.error);
        }
        // return an observable with a user-facing error message
        return Observable.throw({ message: 'Something bad happened; please try again later.' });
    }
    /**
     * @param {?=} query
     * @return {?}
     */
    includeQuery(query = {}) {
        const /** @type {?} */ queryPatch = {};
        Object.keys(query).forEach(key => {
            queryPatch[key] = query[key].toString();
        });
        return queryPatch;
    }
    /**
     * @param {?=} query
     * @return {?}
     */
    includeEncoded(query = {}) {
        const /** @type {?} */ params = new FormData();
        Object.keys(query).forEach((key) => {
            params.append(key, query[key]);
        });
        console.log(params);
        return params;
    }
    /**
     * @param {?} response
     * @param {?=} responseBodyKey
     * @return {?}
     */
    includeResponseHeader(response, responseBodyKey) {
        const /** @type {?} */ headers = {};
        response.headers.keys().forEach(key => {
            headers[key] = response.headers.get(key);
        });
        let /** @type {?} */ responseData = {};
        if (responseBodyKey) {
            responseData[responseBodyKey] = response.body;
        }
        else {
            responseData = response.body;
        }
        responseData['headers'] = headers;
        return responseData;
    }
}
WoocommerceHelperService.decorators = [
    { type: Injectable },
];
/** @nocollapse */
WoocommerceHelperService.ctorParameters = () => [];
function WoocommerceHelperService_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    WoocommerceHelperService.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    WoocommerceHelperService.ctorParameters;
}
