// ngx-wooapi/esm2015/products/product.interface.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * @record
 */
export function Product() { }
function Product_tsickle_Closure_declarations() {
    /** @type {?|undefined} */
    Product.prototype.id;
    /** @type {?|undefined} */
    Product.prototype.name;
    /** @type {?|undefined} */
    Product.prototype.slug;
    /** @type {?|undefined} */
    Product.prototype.permalink;
    /** @type {?|undefined} */
    Product.prototype.date_created;
    /** @type {?|undefined} */
    Product.prototype.date_created_gmt;
    /** @type {?|undefined} */
    Product.prototype.date_modified;
    /** @type {?|undefined} */
    Product.prototype.date_modified_gmt;
    /** @type {?|undefined} */
    Product.prototype.type;
    /** @type {?|undefined} */
    Product.prototype.status;
    /** @type {?|undefined} */
    Product.prototype.featured;
    /** @type {?|undefined} */
    Product.prototype.catalog_visibility;
    /** @type {?|undefined} */
    Product.prototype.description;
    /** @type {?|undefined} */
    Product.prototype.short_description;
    /** @type {?|undefined} */
    Product.prototype.sku;
    /** @type {?|undefined} */
    Product.prototype.price;
    /** @type {?|undefined} */
    Product.prototype.regular_price;
    /** @type {?|undefined} */
    Product.prototype.sale_price;
    /** @type {?|undefined} */
    Product.prototype.date_on_sale_from;
    /** @type {?|undefined} */
    Product.prototype.date_on_sale_from_gmt;
    /** @type {?|undefined} */
    Product.prototype.date_on_sale_to;
    /** @type {?|undefined} */
    Product.prototype.date_on_sale_to_gmt;
    /** @type {?|undefined} */
    Product.prototype.price_html;
    /** @type {?|undefined} */
    Product.prototype.on_sale;
    /** @type {?|undefined} */
    Product.prototype.purchasable;
    /** @type {?|undefined} */
    Product.prototype.total_sales;
    /** @type {?|undefined} */
    Product.prototype.virtual;
    /** @type {?|undefined} */
    Product.prototype.downloadable;
    /** @type {?|undefined} */
    Product.prototype.downloads;
    /** @type {?|undefined} */
    Product.prototype.download_limit;
    /** @type {?|undefined} */
    Product.prototype.download_expiry;
    /** @type {?|undefined} */
    Product.prototype.external_url;
    /** @type {?|undefined} */
    Product.prototype.button_text;
    /** @type {?|undefined} */
    Product.prototype.tax_status;
    /** @type {?|undefined} */
    Product.prototype.tax_class;
    /** @type {?|undefined} */
    Product.prototype.manage_stock;
    /** @type {?|undefined} */
    Product.prototype.stock_quantity;
    /** @type {?|undefined} */
    Product.prototype.in_stock;
    /** @type {?|undefined} */
    Product.prototype.backorders;
    /** @type {?|undefined} */
    Product.prototype.backorders_allowed;
    /** @type {?|undefined} */
    Product.prototype.backordered;
    /** @type {?|undefined} */
    Product.prototype.sold_individually;
    /** @type {?|undefined} */
    Product.prototype.weight;
    /** @type {?|undefined} */
    Product.prototype.dimensions;
    /** @type {?|undefined} */
    Product.prototype.shipping_required;
    /** @type {?|undefined} */
    Product.prototype.shipping_taxable;
    /** @type {?|undefined} */
    Product.prototype.shipping_class;
    /** @type {?|undefined} */
    Product.prototype.shipping_class_id;
    /** @type {?|undefined} */
    Product.prototype.reviews_allowed;
    /** @type {?|undefined} */
    Product.prototype.average_rating;
    /** @type {?|undefined} */
    Product.prototype.rating_count;
    /** @type {?|undefined} */
    Product.prototype.related_ids;
    /** @type {?|undefined} */
    Product.prototype.upsell_ids;
    /** @type {?|undefined} */
    Product.prototype.cross_sell_ids;
    /** @type {?|undefined} */
    Product.prototype.parent_id;
    /** @type {?|undefined} */
    Product.prototype.purchase_note;
    /** @type {?|undefined} */
    Product.prototype.categories;
    /** @type {?|undefined} */
    Product.prototype.tags;
    /** @type {?|undefined} */
    Product.prototype.attributes;
    /** @type {?|undefined} */
    Product.prototype.default_attributes;
    /** @type {?|undefined} */
    Product.prototype.variations;
    /** @type {?|undefined} */
    Product.prototype.grouped_products;
    /** @type {?|undefined} */
    Product.prototype.menu_order;
    /** @type {?|undefined} */
    Product.prototype.meta_data;
}
/**
 * @record
 */
export function RetrieveProductsResponse() { }
function RetrieveProductsResponse_tsickle_Closure_declarations() {
    /** @type {?|undefined} */
    RetrieveProductsResponse.prototype.products;
    /** @type {?|undefined} */
    RetrieveProductsResponse.prototype.headers;
}
/** @enum {string} */
const ProductOrder = {
    "desc": 'desc',
    "asc": 'asc',
};
export { ProductOrder };
/** @enum {string} */
const ProductOrderBy = {
    "date": 'date',
    "id": 'id',
    "include": 'include',
    "title": 'title',
    "slug": 'slug',
};
export { ProductOrderBy };
/**
 * @record
 */
export function ProductQuery() { }
function ProductQuery_tsickle_Closure_declarations() {
    /** @type {?|undefined} */
    ProductQuery.prototype.context;
    /** @type {?|undefined} */
    ProductQuery.prototype.page;
    /** @type {?|undefined} */
    ProductQuery.prototype.per_page;
    /** @type {?|undefined} */
    ProductQuery.prototype.search;
    /** @type {?|undefined} */
    ProductQuery.prototype.after;
    /** @type {?|undefined} */
    ProductQuery.prototype.before;
    /** @type {?|undefined} */
    ProductQuery.prototype.exclude;
    /** @type {?|undefined} */
    ProductQuery.prototype.include;
    /** @type {?|undefined} */
    ProductQuery.prototype.offset;
    /** @type {?|undefined} */
    ProductQuery.prototype.order;
    /** @type {?|undefined} */
    ProductQuery.prototype.orderby;
    /** @type {?|undefined} */
    ProductQuery.prototype.parent;
    /** @type {?|undefined} */
    ProductQuery.prototype.parent_exclude;
    /** @type {?|undefined} */
    ProductQuery.prototype.slug;
    /** @type {?|undefined} */
    ProductQuery.prototype.status;
    /** @type {?|undefined} */
    ProductQuery.prototype.type;
    /** @type {?|undefined} */
    ProductQuery.prototype.sku;
    /** @type {?|undefined} */
    ProductQuery.prototype.featured;
    /** @type {?|undefined} */
    ProductQuery.prototype.category;
    /** @type {?|undefined} */
    ProductQuery.prototype.tag;
    /** @type {?|undefined} */
    ProductQuery.prototype.shipping_class;
    /** @type {?|undefined} */
    ProductQuery.prototype.attribute;
    /** @type {?|undefined} */
    ProductQuery.prototype.attribute_term;
    /** @type {?|undefined} */
    ProductQuery.prototype.tax_class;
    /** @type {?|undefined} */
    ProductQuery.prototype.in_stock;
    /** @type {?|undefined} */
    ProductQuery.prototype.on_sale;
    /** @type {?|undefined} */
    ProductQuery.prototype.min_price;
    /** @type {?|undefined} */
    ProductQuery.prototype.max_price;
}
/**
 * @record
 */
export function ProductReview() { }
function ProductReview_tsickle_Closure_declarations() {
    /** @type {?|undefined} */
    ProductReview.prototype.id;
    /** @type {?|undefined} */
    ProductReview.prototype.review;
    /** @type {?|undefined} */
    ProductReview.prototype.date_created;
    /** @type {?|undefined} */
    ProductReview.prototype.date_created_gmt;
    /** @type {?|undefined} */
    ProductReview.prototype.rating;
    /** @type {?|undefined} */
    ProductReview.prototype.name;
    /** @type {?|undefined} */
    ProductReview.prototype.email;
    /** @type {?|undefined} */
    ProductReview.prototype.verified;
}
/**
 * @record
 */
export function ProductReviewsResponse() { }
function ProductReviewsResponse_tsickle_Closure_declarations() {
    /** @type {?|undefined} */
    ProductReviewsResponse.prototype.product_reviews;
}
