// ngx-wooapi/esm2015/products/woocommerce-products.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { catchError, map } from 'rxjs/operators';
import { WoocommerceHelperService } from '../helper.service';
export class WoocommerceProductsService {
    /**
     * @param {?} httpClient
     * @param {?} wooHelper
     */
    constructor(httpClient, wooHelper) {
        this.httpClient = httpClient;
        this.wooHelper = wooHelper;
    }
    /**
     * Create a Product
     * @param {?} payload
     * @return {?}
     */
    createProduct(payload) {
        return this.httpClient.post(`products`, payload)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Retrieve a product
     * @param {?} id
     * @return {?}
     */
    retrieveProduct(id) {
        return this.httpClient.get(`products/${id}`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Retrieve list of product
     * @param {?=} query
     * @return {?}
     */
    retrieveProducts(query) {
        return this.httpClient.get(`products`, { params: this.wooHelper.includeQuery(query), observe: 'response' })
            .pipe(map(value => this.wooHelper.includeResponseHeader(value, 'products')), catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Update Product
     * @param {?} id
     * @param {?} payload
     * @return {?}
     */
    updateProduct(id, payload) {
        return this.httpClient.put(`products/${id}`, payload)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Update Product
     * @param {?} id
     * @return {?}
     */
    deleteProduct(id) {
        return this.httpClient.delete(`products/${id}`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Retrieve product reviews by product id
     * @param {?} product_id
     * @return {?}
     */
    retrieveProductReviews(product_id) {
        return this.httpClient.get(`products/${product_id}/reviews`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Retrieve product reviews by product id
     * @param {?} product_id
     * @param {?} review_id
     * @return {?}
     */
    retrieveProductReview(product_id, review_id) {
        return this.httpClient.get(`products/${product_id}/reviews/${review_id}`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
}
WoocommerceProductsService.decorators = [
    { type: Injectable },
];
/** @nocollapse */
WoocommerceProductsService.ctorParameters = () => [
    { type: HttpClient, },
    { type: WoocommerceHelperService, },
];
function WoocommerceProductsService_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    WoocommerceProductsService.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    WoocommerceProductsService.ctorParameters;
    /** @type {?} */
    WoocommerceProductsService.prototype.httpClient;
    /** @type {?} */
    WoocommerceProductsService.prototype.wooHelper;
}
