// ngx-wooapi/esm2015/customer/customer.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { catchError } from 'rxjs/operators';
import { WoocommerceHelperService } from '../helper.service';
export class WoocommerceCustomerService {
    /**
     * @param {?} httpClient
     * @param {?} wooHelper
     */
    constructor(httpClient, wooHelper) {
        this.httpClient = httpClient;
        this.wooHelper = wooHelper;
    }
    /**
     * Create a customer
     * @param {?} customer
     * @return {?}
     */
    createCustomers(customer) {
        return this.httpClient.post(`customers`, customer)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Retrive a customer
     * @param {?} id
     * @return {?}
     */
    retriveCustomers(id) {
        return this.httpClient.get(`customers/${id}`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Update a customer
     * @param {?} id
     * @param {?} customer
     * @return {?}
     */
    updateCustomers(id, customer) {
        return this.httpClient.put(`customers/${id}`, customer)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
}
WoocommerceCustomerService.decorators = [
    { type: Injectable },
];
/** @nocollapse */
WoocommerceCustomerService.ctorParameters = () => [
    { type: HttpClient, },
    { type: WoocommerceHelperService, },
];
function WoocommerceCustomerService_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    WoocommerceCustomerService.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    WoocommerceCustomerService.ctorParameters;
    /** @type {?} */
    WoocommerceCustomerService.prototype.httpClient;
    /** @type {?} */
    WoocommerceCustomerService.prototype.wooHelper;
}
