// ngx-wooapi/esm2015/orders/orders.interface.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * @record
 */
export function Order() { }
function Order_tsickle_Closure_declarations() {
    /** @type {?|undefined} */
    Order.prototype.id;
    /** @type {?|undefined} */
    Order.prototype.parent_id;
    /** @type {?|undefined} */
    Order.prototype.number;
    /** @type {?|undefined} */
    Order.prototype.order_key;
    /** @type {?|undefined} */
    Order.prototype.created_via;
    /** @type {?|undefined} */
    Order.prototype.version;
    /** @type {?|undefined} */
    Order.prototype.status;
    /** @type {?|undefined} */
    Order.prototype.currency;
    /** @type {?|undefined} */
    Order.prototype.date_created;
    /** @type {?|undefined} */
    Order.prototype.date_created_gmt;
    /** @type {?|undefined} */
    Order.prototype.date_modified;
    /** @type {?|undefined} */
    Order.prototype.date_modified_gmt;
    /** @type {?|undefined} */
    Order.prototype.discount_total;
    /** @type {?|undefined} */
    Order.prototype.discount_tax;
    /** @type {?|undefined} */
    Order.prototype.shipping_total;
    /** @type {?|undefined} */
    Order.prototype.shipping_tax;
    /** @type {?|undefined} */
    Order.prototype.cart_tax;
    /** @type {?|undefined} */
    Order.prototype.total;
    /** @type {?|undefined} */
    Order.prototype.total_tax;
    /** @type {?|undefined} */
    Order.prototype.prices_include_tax;
    /** @type {?|undefined} */
    Order.prototype.customer_id;
    /** @type {?|undefined} */
    Order.prototype.customer_ip_address;
    /** @type {?|undefined} */
    Order.prototype.customer_user_agent;
    /** @type {?|undefined} */
    Order.prototype.customer_note;
    /** @type {?|undefined} */
    Order.prototype.billing;
    /** @type {?|undefined} */
    Order.prototype.shipping;
    /** @type {?|undefined} */
    Order.prototype.payment_method;
    /** @type {?|undefined} */
    Order.prototype.payment_method_title;
    /** @type {?|undefined} */
    Order.prototype.transaction_id;
    /** @type {?|undefined} */
    Order.prototype.date_paid;
    /** @type {?|undefined} */
    Order.prototype.date_paid_gmt;
    /** @type {?|undefined} */
    Order.prototype.date_completed;
    /** @type {?|undefined} */
    Order.prototype.date_completed_gmt;
    /** @type {?|undefined} */
    Order.prototype.cart_hash;
    /** @type {?|undefined} */
    Order.prototype.meta_data;
    /** @type {?|undefined} */
    Order.prototype.line_items;
    /** @type {?|undefined} */
    Order.prototype.tax_lines;
    /** @type {?|undefined} */
    Order.prototype.shipping_lines;
    /** @type {?|undefined} */
    Order.prototype.fee_lines;
    /** @type {?|undefined} */
    Order.prototype.coupon_lines;
    /** @type {?|undefined} */
    Order.prototype.refunds;
    /** @type {?|undefined} */
    Order.prototype.set_paid;
}
/**
 * @record
 */
export function Billing() { }
function Billing_tsickle_Closure_declarations() {
    /** @type {?|undefined} */
    Billing.prototype.last_name;
    /** @type {?|undefined} */
    Billing.prototype.company;
    /** @type {?|undefined} */
    Billing.prototype.address_1;
    /** @type {?|undefined} */
    Billing.prototype.address_2;
    /** @type {?|undefined} */
    Billing.prototype.city;
    /** @type {?|undefined} */
    Billing.prototype.state;
    /** @type {?|undefined} */
    Billing.prototype.postcode;
    /** @type {?|undefined} */
    Billing.prototype.country;
    /** @type {?|undefined} */
    Billing.prototype.email;
    /** @type {?|undefined} */
    Billing.prototype.phone;
}
/**
 * @record
 */
export function Shipping() { }
function Shipping_tsickle_Closure_declarations() {
    /** @type {?|undefined} */
    Shipping.prototype.last_name;
    /** @type {?|undefined} */
    Shipping.prototype.company;
    /** @type {?|undefined} */
    Shipping.prototype.address_1;
    /** @type {?|undefined} */
    Shipping.prototype.address_2;
    /** @type {?|undefined} */
    Shipping.prototype.city;
    /** @type {?|undefined} */
    Shipping.prototype.state;
    /** @type {?|undefined} */
    Shipping.prototype.postcode;
    /** @type {?|undefined} */
    Shipping.prototype.country;
}
/**
 * @record
 */
export function OrderItem() { }
function OrderItem_tsickle_Closure_declarations() {
    /** @type {?} */
    OrderItem.prototype.name;
    /** @type {?} */
    OrderItem.prototype.product_id;
    /** @type {?} */
    OrderItem.prototype.variation_id;
    /** @type {?} */
    OrderItem.prototype.quantity;
    /** @type {?} */
    OrderItem.prototype.tax_class;
    /** @type {?} */
    OrderItem.prototype.subtotal;
    /** @type {?} */
    OrderItem.prototype.subtotal_tax;
    /** @type {?} */
    OrderItem.prototype.total;
    /** @type {?} */
    OrderItem.prototype.total_tax;
    /** @type {?} */
    OrderItem.prototype.taxes;
    /** @type {?} */
    OrderItem.prototype.meta_data;
    /** @type {?} */
    OrderItem.prototype.sku;
    /** @type {?} */
    OrderItem.prototype.price;
}
/**
 * @record
 */
export function OrderTax() { }
function OrderTax_tsickle_Closure_declarations() {
    /** @type {?} */
    OrderTax.prototype.rate_code;
    /** @type {?} */
    OrderTax.prototype.rate_id;
    /** @type {?} */
    OrderTax.prototype.label;
    /** @type {?} */
    OrderTax.prototype.compound;
    /** @type {?} */
    OrderTax.prototype.tax_total;
    /** @type {?} */
    OrderTax.prototype.shipping_tax_total;
    /** @type {?} */
    OrderTax.prototype.meta_data;
}
