// ngx-wooapi/esm2015/orders/order.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { catchError } from 'rxjs/operators';
import { WoocommerceHelperService } from '../helper.service';
export class WoocommerceOrderService {
    /**
     * @param {?} httpClient
     * @param {?} wooHelper
     */
    constructor(httpClient, wooHelper) {
        this.httpClient = httpClient;
        this.wooHelper = wooHelper;
    }
    /**
     * @param {?} order
     * @return {?}
     */
    createOrder(order) {
        return this.httpClient.post(`orders`, order)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * @param {?} id
     * @return {?}
     */
    retriveOrder(id) {
        return this.httpClient.get(`orders/${id}`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * @return {?}
     */
    listAllOrder() {
        return this.httpClient.get(`orders`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * @param {?} order
     * @return {?}
     */
    updateOrder(order) {
        return this.httpClient.put(`orders/${order.id}`, order)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * @param {?} id
     * @return {?}
     */
    deleteOrder(id) {
        return this.httpClient.delete(`orders/${id}`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
}
WoocommerceOrderService.decorators = [
    { type: Injectable },
];
/** @nocollapse */
WoocommerceOrderService.ctorParameters = () => [
    { type: HttpClient, },
    { type: WoocommerceHelperService, },
];
function WoocommerceOrderService_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    WoocommerceOrderService.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    WoocommerceOrderService.ctorParameters;
    /** @type {?} */
    WoocommerceOrderService.prototype.httpClient;
    /** @type {?} */
    WoocommerceOrderService.prototype.wooHelper;
}
