// ngx-wooapi/esm5/wooApi.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
export { WoocommerceHelperService } from './helper.service';
export { ProductOrder, ProductOrderBy } from './products/product.interface';
export { WoocommerceProductsService } from './products/woocommerce-products.service';
export { TagOrder, TagOrderBy } from './tags/product-tags.interface';
export { WoocommerceTagsService } from './tags/woocommerce-tags.service';
export { CategoryOrder, CategoryOrderBy } from './categories/product-categories.interface';
export { WoocommerceCategoriesService } from './categories/woocommerce-categories.service';
export { NonceMethod } from './auth/auth.interface';
export { AuthService } from './auth/auth.service';
export { WoocommerceOrderService } from './orders/order.service';
export { WoocommerceCouponService } from './coupon/coupon.service';
export { WoocommerceCustomerService } from './customer/customer.service';
export { WoocommerceOrderNoteService } from './order-notes/order-notes.service';
