// ngx-wooapi/esm5/categories/product-categories.interface.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * @record
 */
export function ProductCategory() { }
function ProductCategory_tsickle_Closure_declarations() {
    /** @type {?|undefined} */
    ProductCategory.prototype.id;
    /** @type {?|undefined} */
    ProductCategory.prototype.name;
    /** @type {?|undefined} */
    ProductCategory.prototype.slug;
    /** @type {?|undefined} */
    ProductCategory.prototype.parent;
    /** @type {?|undefined} */
    ProductCategory.prototype.description;
    /** @type {?|undefined} */
    ProductCategory.prototype.display;
    /** @type {?|undefined} */
    ProductCategory.prototype.image;
    /** @type {?|undefined} */
    ProductCategory.prototype.menu_order;
    /** @type {?|undefined} */
    ProductCategory.prototype.count;
}
/**
 * @record
 */
export function ProductCategoryImage() { }
function ProductCategoryImage_tsickle_Closure_declarations() {
    /** @type {?|undefined} */
    ProductCategoryImage.prototype.id;
    /** @type {?|undefined} */
    ProductCategoryImage.prototype.date_created;
    /** @type {?|undefined} */
    ProductCategoryImage.prototype.date_created_gmt;
    /** @type {?|undefined} */
    ProductCategoryImage.prototype.date_modified;
    /** @type {?|undefined} */
    ProductCategoryImage.prototype.date_modified_gmt;
    /** @type {?|undefined} */
    ProductCategoryImage.prototype.src;
    /** @type {?|undefined} */
    ProductCategoryImage.prototype.title;
    /** @type {?|undefined} */
    ProductCategoryImage.prototype.alt;
}
/** @enum {string} */
var CategoryOrder = {
    "desc": 'desc',
    "asc": 'asc',
};
export { CategoryOrder };
/** @enum {string} */
var CategoryOrderBy = {
    "id": 'id',
    "include": 'include',
    "name": 'name',
    "slug": 'slug',
    "term_group": 'term_group',
    "description": 'description',
    "count": 'count',
};
export { CategoryOrderBy };
/**
 * @record
 */
export function CategoryQuery() { }
function CategoryQuery_tsickle_Closure_declarations() {
    /** @type {?|undefined} */
    CategoryQuery.prototype.context;
    /** @type {?|undefined} */
    CategoryQuery.prototype.page;
    /** @type {?|undefined} */
    CategoryQuery.prototype.per_page;
    /** @type {?|undefined} */
    CategoryQuery.prototype.search;
    /** @type {?|undefined} */
    CategoryQuery.prototype.exclude;
    /** @type {?|undefined} */
    CategoryQuery.prototype.include;
    /** @type {?|undefined} */
    CategoryQuery.prototype.order;
    /** @type {?|undefined} */
    CategoryQuery.prototype.orderby;
    /** @type {?|undefined} */
    CategoryQuery.prototype.hide_empty;
    /** @type {?|undefined} */
    CategoryQuery.prototype.parent;
    /** @type {?|undefined} */
    CategoryQuery.prototype.product;
    /** @type {?|undefined} */
    CategoryQuery.prototype.slug;
}
