// ngx-wooapi/esm5/categories/woocommerce-categories.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { catchError } from 'rxjs/operators';
import { WoocommerceHelperService } from '../helper.service';
var WoocommerceCategoriesService = /** @class */ (function () {
    function WoocommerceCategoriesService(httpClient, wooHelper) {
        this.httpClient = httpClient;
        this.wooHelper = wooHelper;
    }
    /**
     * Create a Category
     * @param payload: Product
     */
    /**
     * Create a Category
     * @param {?} category
     * @return {?}
     */
    WoocommerceCategoriesService.prototype.createCategory = /**
     * Create a Category
     * @param {?} category
     * @return {?}
     */
    function (category) {
        var _this = this;
        return this.httpClient.post("products/categories", category)
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    /**
     * Retrieve a Category
     */
    /**
     * Retrieve a Category
     * @param {?} id
     * @return {?}
     */
    WoocommerceCategoriesService.prototype.retrieveCategory = /**
     * Retrieve a Category
     * @param {?} id
     * @return {?}
     */
    function (id) {
        var _this = this;
        return this.httpClient.get("products/categories/" + id)
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    /**
     * Retrieve list of Category
     */
    /**
     * Retrieve list of Category
     * @param {?=} query
     * @return {?}
     */
    WoocommerceCategoriesService.prototype.retrieveCategories = /**
     * Retrieve list of Category
     * @param {?=} query
     * @return {?}
     */
    function (query) {
        var _this = this;
        return this.httpClient.get("products/categories", {
            params: this.wooHelper.includeQuery(query)
        })
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    /**
     * Update Category
     */
    /**
     * Update Category
     * @param {?} id
     * @param {?} payload
     * @return {?}
     */
    WoocommerceCategoriesService.prototype.updateCategory = /**
     * Update Category
     * @param {?} id
     * @param {?} payload
     * @return {?}
     */
    function (id, payload) {
        var _this = this;
        return this.httpClient.put("products/categories/" + id, payload)
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    /**
     * Delete Category
     */
    /**
     * Delete Category
     * @param {?} id
     * @return {?}
     */
    WoocommerceCategoriesService.prototype.deleteCategory = /**
     * Delete Category
     * @param {?} id
     * @return {?}
     */
    function (id) {
        var _this = this;
        return this.httpClient.delete("products/categories/" + id)
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    WoocommerceCategoriesService.decorators = [
        { type: Injectable },
    ];
    /** @nocollapse */
    WoocommerceCategoriesService.ctorParameters = function () { return [
        { type: HttpClient, },
        { type: WoocommerceHelperService, },
    ]; };
    return WoocommerceCategoriesService;
}());
export { WoocommerceCategoriesService };
function WoocommerceCategoriesService_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    WoocommerceCategoriesService.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    WoocommerceCategoriesService.ctorParameters;
    /** @type {?} */
    WoocommerceCategoriesService.prototype.httpClient;
    /** @type {?} */
    WoocommerceCategoriesService.prototype.wooHelper;
}
