// ngx-wooapi/esm5/orders/order.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { catchError } from 'rxjs/operators';
import { WoocommerceHelperService } from '../helper.service';
var WoocommerceOrderService = /** @class */ (function () {
    function WoocommerceOrderService(httpClient, wooHelper) {
        this.httpClient = httpClient;
        this.wooHelper = wooHelper;
    }
    /**
     * @param {?} order
     * @return {?}
     */
    WoocommerceOrderService.prototype.createOrder = /**
     * @param {?} order
     * @return {?}
     */
    function (order) {
        var _this = this;
        return this.httpClient.post("orders", order)
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    /**
     * @param {?} id
     * @return {?}
     */
    WoocommerceOrderService.prototype.retriveOrder = /**
     * @param {?} id
     * @return {?}
     */
    function (id) {
        var _this = this;
        return this.httpClient.get("orders/" + id)
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    /**
     * @return {?}
     */
    WoocommerceOrderService.prototype.listAllOrder = /**
     * @return {?}
     */
    function () {
        var _this = this;
        return this.httpClient.get("orders")
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    /**
     * @param {?} order
     * @return {?}
     */
    WoocommerceOrderService.prototype.updateOrder = /**
     * @param {?} order
     * @return {?}
     */
    function (order) {
        var _this = this;
        return this.httpClient.put("orders/" + order.id, order)
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    /**
     * @param {?} id
     * @return {?}
     */
    WoocommerceOrderService.prototype.deleteOrder = /**
     * @param {?} id
     * @return {?}
     */
    function (id) {
        var _this = this;
        return this.httpClient.delete("orders/" + id)
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    WoocommerceOrderService.decorators = [
        { type: Injectable },
    ];
    /** @nocollapse */
    WoocommerceOrderService.ctorParameters = function () { return [
        { type: HttpClient, },
        { type: WoocommerceHelperService, },
    ]; };
    return WoocommerceOrderService;
}());
export { WoocommerceOrderService };
function WoocommerceOrderService_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    WoocommerceOrderService.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    WoocommerceOrderService.ctorParameters;
    /** @type {?} */
    WoocommerceOrderService.prototype.httpClient;
    /** @type {?} */
    WoocommerceOrderService.prototype.wooHelper;
}
