// ngx-wooapi/esm2015/categories/woocommerce-categories.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { catchError } from 'rxjs/operators';
import { WoocommerceHelperService } from '../helper.service';
export class WoocommerceCategoriesService {
    /**
     * @param {?} httpClient
     * @param {?} wooHelper
     */
    constructor(httpClient, wooHelper) {
        this.httpClient = httpClient;
        this.wooHelper = wooHelper;
    }
    /**
     * Create a Category
     * @param {?} category
     * @return {?}
     */
    createCategory(category) {
        return this.httpClient.post(`products/categories`, category)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Retrieve a Category
     * @param {?} id
     * @return {?}
     */
    retrieveCategory(id) {
        return this.httpClient.get(`products/categories/${id}`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Retrieve list of Category
     * @param {?=} query
     * @return {?}
     */
    retrieveCategories(query) {
        return this.httpClient.get(`products/categories`, {
            params: this.wooHelper.includeQuery(query)
        })
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Update Category
     * @param {?} id
     * @param {?} payload
     * @return {?}
     */
    updateCategory(id, payload) {
        return this.httpClient.put(`products/categories/${id}`, payload)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
    /**
     * Delete Category
     * @param {?} id
     * @return {?}
     */
    deleteCategory(id) {
        return this.httpClient.delete(`products/categories/${id}`)
            .pipe(catchError(err => this.wooHelper.handleError(err)));
    }
}
WoocommerceCategoriesService.decorators = [
    { type: Injectable },
];
/** @nocollapse */
WoocommerceCategoriesService.ctorParameters = () => [
    { type: HttpClient, },
    { type: WoocommerceHelperService, },
];
function WoocommerceCategoriesService_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    WoocommerceCategoriesService.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    WoocommerceCategoriesService.ctorParameters;
    /** @type {?} */
    WoocommerceCategoriesService.prototype.httpClient;
    /** @type {?} */
    WoocommerceCategoriesService.prototype.wooHelper;
}
