// ngx-wooapi/esm5/order-notes/order-notes.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { catchError } from 'rxjs/operators';
import { WoocommerceHelperService } from '../helper.service';
var WoocommerceOrderNoteService = /** @class */ (function () {
    function WoocommerceOrderNoteService(httpClient, wooHelper) {
        this.httpClient = httpClient;
        this.wooHelper = wooHelper;
    }
    /**
     * Create a order note
     * URL: https://woocommerce.github.io/woocommerce-rest-api-docs/#create-an-order-note
     */
    /**
     * Create a order note
     * URL: https://woocommerce.github.io/woocommerce-rest-api-docs/#create-an-order-note
     * @param {?} id
     * @param {?} note
     * @return {?}
     */
    WoocommerceOrderNoteService.prototype.createNote = /**
     * Create a order note
     * URL: https://woocommerce.github.io/woocommerce-rest-api-docs/#create-an-order-note
     * @param {?} id
     * @param {?} note
     * @return {?}
     */
    function (id, note) {
        var _this = this;
        return this.httpClient.post("orders/" + id + "/notes", note)
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    /**
     * Retrive a order note
     * URL: https://woocommerce.github.io/woocommerce-rest-api-docs/#retrieve-an-order-note
     */
    /**
     * Retrive a order note
     * URL: https://woocommerce.github.io/woocommerce-rest-api-docs/#retrieve-an-order-note
     * @param {?} id
     * @param {?} noteid
     * @return {?}
     */
    WoocommerceOrderNoteService.prototype.retriveNote = /**
     * Retrive a order note
     * URL: https://woocommerce.github.io/woocommerce-rest-api-docs/#retrieve-an-order-note
     * @param {?} id
     * @param {?} noteid
     * @return {?}
     */
    function (id, noteid) {
        var _this = this;
        return this.httpClient.get("orders/" + id + "/notes/" + noteid)
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    /**
    * List all order notes
    * URL: https://woocommerce.github.io/woocommerce-rest-api-docs/#list-all-order-notes
    */
    /**
     * List all order notes
     * URL: https://woocommerce.github.io/woocommerce-rest-api-docs/#list-all-order-notes
     * @param {?} id
     * @return {?}
     */
    WoocommerceOrderNoteService.prototype.retriveNotes = /**
     * List all order notes
     * URL: https://woocommerce.github.io/woocommerce-rest-api-docs/#list-all-order-notes
     * @param {?} id
     * @return {?}
     */
    function (id) {
        var _this = this;
        return this.httpClient.get("orders/" + id + "/notes/")
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    /**
     * Delete a order note
     * URL: https://woocommerce.github.io/woocommerce-rest-api-docs/#delete-an-order-note
     */
    /**
     * Delete a order note
     * URL: https://woocommerce.github.io/woocommerce-rest-api-docs/#delete-an-order-note
     * @param {?} id
     * @param {?} noteid
     * @return {?}
     */
    WoocommerceOrderNoteService.prototype.deleteNote = /**
     * Delete a order note
     * URL: https://woocommerce.github.io/woocommerce-rest-api-docs/#delete-an-order-note
     * @param {?} id
     * @param {?} noteid
     * @return {?}
     */
    function (id, noteid) {
        var _this = this;
        return this.httpClient.delete("orders/" + id + "/notes/" + noteid)
            .pipe(catchError(function (err) { return _this.wooHelper.handleError(err); }));
    };
    WoocommerceOrderNoteService.decorators = [
        { type: Injectable },
    ];
    /** @nocollapse */
    WoocommerceOrderNoteService.ctorParameters = function () { return [
        { type: HttpClient, },
        { type: WoocommerceHelperService, },
    ]; };
    return WoocommerceOrderNoteService;
}());
export { WoocommerceOrderNoteService };
function WoocommerceOrderNoteService_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    WoocommerceOrderNoteService.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    WoocommerceOrderNoteService.ctorParameters;
    /** @type {?} */
    WoocommerceOrderNoteService.prototype.httpClient;
    /** @type {?} */
    WoocommerceOrderNoteService.prototype.wooHelper;
}
